.github/workflows/upgrade-python-requirements.yml

```diff
@@ -0,0 +1,71 @@
+name: Upgrade Requirements
+
+on:
+  push:
+    branches:
+      - '**'
+  schedule:
+    # will start the job at 3:15 UTC every Thursday
+    - cron: "15 3 * * 4"
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: "Target branch to create requirements PR against"
+        required: true
+        default: 'master'
+
+jobs:
+  upgrade_requirements:
+    runs-on: ubuntu-20.04
+
+    strategy:
+      matrix:
+        python-version: ["3.8"]
+
+    steps:
+      - name: setup target branch
+        run: echo "target_branch=$(if ['${{ github.event.inputs.branch }}' = '']; then echo 'master'; else echo '${{ github.event.inputs.branch }}'; fi)" >> $GITHUB_ENV
+
+      - uses: actions/checkout@v1
+        with:
+          ref: ${{ env.target_branch }}
+      
+      - name: setup python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: make upgrade
+        run: |
+          cd $GITHUB_WORKSPACE
+          make upgrade
+
+      - name: setup testeng-ci
+        run: |
+          git clone https://github.com/edx/testeng-ci.git
+          cd $GITHUB_WORKSPACE/testeng-ci
+          pip install -r requirements/base.txt
+      - name: create pull request
+        env:
+          GITHUB_TOKEN: ${{ secrets.REQUIREMENTS_BOT_GITHUB_TOKEN }}
+          GITHUB_USER_EMAIL: ${{ secrets.REQUIREMENTS_BOT_GITHUB_EMAIL }}
+        run: |  
+          cd $GITHUB_WORKSPACE/testeng-ci
+          python -m jenkins.pull_request_creator --repo-root=$GITHUB_WORKSPACE \
+          --target-branch="${{ env.target_branch }}" --base-branch-name="upgrade-python-requirements" \
+          --commit-message="chore: Updating Python Requirements" --pr-title="Python Requirements Update" \
+          --pr-body="Python requirements update.Please review the [changelogs](https://openedx.atlassian.net/wiki/spaces/TE/pages/1001521320/Python+Package+Changelogs) for the upgraded packages." \
+          --user-reviewers="" --team-reviewers="masters-devs-cosmonauts" --delete-old-pull-requests
+
+      - name: Send failure notification
+        if: ${{ failure() }}
```

Comment:
here as elsewhere

we have our notifications set to send an email whether or not our job fails; this enables us to route an opsgenie alert to whoever's responsible for doing this chore on a particular week. See https://github.com/edx/jenkins-job-dsl/blob/32b331a9c1a3b2683e3bd7e191062cc72b0f1094/testeng/jobs/upgradePythonRequirements.groovy#L277 for how we had that set up in Jenkins. It would be better if we didn't have to regress on this.